src/pgtz/src/timezone.rs

```diff
@@ -0,0 +1,497 @@
+        Ok(match kind {
+            Kind::FixedOffset(pof) => Timezone::FixedOffset(FixedOffset::from_proto(pof)?),
+            Kind::Tz(ptz) => Timezone::Tz(Tz::from_proto(ptz)?),
+        })
+    }
+}
+
+// We need to implement Serialize and Deserialize traits to include Timezone in the UnaryFunc enum.
+// FixedOffset doesn't implement these, even with the "serde" feature enabled.
+mod fixed_offset_serde {
+    use serde::de::Error;
+    use serde::{Deserializer, Serializer};
+
+    use super::*;
+
+    pub fn deserialize<'de, D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<FixedOffset, D::Error> {
+        let offset = i32::deserialize(deserializer)?;
+        FixedOffset::east_opt(offset).ok_or_else(|| {
+            Error::custom(format!("Invalid timezone offset: |{}| >= 86_400", offset))
+        })
+    }
+
+    pub fn serialize<S: Serializer>(
+        offset: &FixedOffset,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        serializer.serialize_i32(offset.local_minus_utc())
+    }
+}
+
+impl PartialOrd for Timezone {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+// We need to implement Ord and PartialOrd to include Timezone in the UnaryFunc enum. Neither FixedOffset nor Tz
+// implement these so we do a simple ordinal comparison (FixedOffset variant < Tz variant), and break ties using
+// i32/str comparisons respectively.
+impl Ord for Timezone {
+    fn cmp(&self, other: &Self) -> Ordering {
+        use Timezone::*;
+        match (self, other) {
+            (FixedOffset(a), FixedOffset(b)) => a.local_minus_utc().cmp(&b.local_minus_utc()),
+            (Tz(a), Tz(b)) => a.name().cmp(b.name()),
+            (FixedOffset(_), Tz(_)) => Ordering::Less,
+            (Tz(_), FixedOffset(_)) => Ordering::Greater,
+        }
+    }
+}
+
+impl Default for Timezone {
+    fn default() -> Self {
+        Self::FixedOffset(FixedOffset::east_opt(0).unwrap())
+    }
+}
+
+impl fmt::Display for Timezone {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Timezone::FixedOffset(offset) => offset.fmt(f),
+            Timezone::Tz(tz) => tz.fmt(f),
+        }
+    }
+}
+
+/// TimeStrToken represents valid tokens in time-like strings,
+/// i.e those used in INTERVAL, TIMESTAMP/TZ, DATE, and TIME.
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum TimeStrToken {
+    Dash,
+    Colon,
+    Plus,
+    Zulu,
+    Num(u64, usize),
+    TzName(String),
+    Delim,
+}
+
+impl std::fmt::Display for TimeStrToken {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use TimeStrToken::*;
+        match self {
+            Dash => write!(f, "-"),
+            Colon => write!(f, ":"),
+            Plus => write!(f, "+"),
+            Zulu => write!(f, "Z"),
+            Num(i, digits) => write!(f, "{:01$}", i, digits - 1),
+            TzName(n) => write!(f, "{}", n),
+            Delim => write!(f, " "),
```

Comment:
Untested according to coverage report.

Reply:
This was code movement, not going to add a test at the moment.